File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int globalMax = 0;
        int left = 0;
        int right = height.length - 1;

        while (left < right) {
            int area = (right - left) * Math.min(height[left], height[right]);
            globalMax = Math.max(globalMax, area);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return globalMax;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int result = 0;
        while (left < right) {
            if (height[left] < height[right]) {
                int len = height[left];
                int width = right - left;
                result = max(result, len * width);
                left++;
            } else {
                int len = height[right];
                int width = right - left;
                result = max(result, len * width);
                right--;                
            }
        }

        return result;
    }
};